src/common/src/array/arrow/arrow_impl.rs

```diff
@@ -522,6 +522,7 @@ pub trait FromArrow {
             Int16 => self.from_int16_array(array.as_any().downcast_ref().unwrap()),
             Int32 => self.from_int32_array(array.as_any().downcast_ref().unwrap()),
             Int64 => self.from_int64_array(array.as_any().downcast_ref().unwrap()),
+            Decimal128(_, _) => self.from_decimal128_array(array.as_any().downcast_ref().unwrap()),
             Decimal256(_, _) => self.from_int256_array(array.as_any().downcast_ref().unwrap()),
```

Comment:
why decimal 128 and 256 are different? It is from_int256 below 🤔

Reply:
seems a typo for 256

Reply:
Arrow decimal256 type is used for RisingWave int256 type. 🤔